api/dp-openapi.yaml

```diff
@@ -0,0 +1,245 @@
+      - name: projectid
+        in: path
+        description: ID of project to return
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: project response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/project'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '429':
+          $ref: '#/components/responses/RateLimitExceeded'
+        default:
+          $ref: '#/components/responses/UnexpectedError'
+
+
+  /projects/{projectid}/wordlists/{type}:
+    get:
+      tags:
+      - project
+      description: Gets a project's word list
+      parameters:
+      - name: projectid
+        in: path
+        description: ID of project
+        required: true
+        schema:
+          type: string
+      - name: type
+        in: path
+        description: Word list type
+        required: true
+        schema:
+          type: string
+          enum: [good, bad]
+      responses:
+        200:
+          description: Requested project word list
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/wordlist'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '429':
+          $ref: '#/components/responses/RateLimitExceeded'
+        default:
+          $ref: '#/components/responses/UnexpectedError'
+    put:
+      tags:
+      - project
+      description: Replaces a project's word list
+      parameters:
+      - name: projectid
+        in: path
+        description: ID of project
+        required: true
+        schema:
+          type: string
+      - name: type
+        in: path
+        description: Word list type
+        required: true
+        schema:
+          type: string
+          enum: [good, bad]
+      requestBody:
+        description: Words to replace
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/wordlist'
+        required: true
+      responses:
+        200:
+          description: wordlist saved
+          content: {}
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '429':
+          $ref: '#/components/responses/RateLimitExceeded'
+        default:
+          $ref: '#/components/responses/UnexpectedError'
+
+components:
+  securitySchemes:
+    ApiKeyAuth:
+      type: apiKey
+      in: header
+      name: X-API-KEY
+
+  schemas:
+    project:
+      required:
+      - projectid
+      - title
+      type: object
+      properties:
+        projectid:
+          type: string
+        title:
+          type: string
+        author:
+          type: string
+        languages:
```

Comment:
A nit here, but I find the YAML for the property types really verbose and hard to eyeball. YAML has an alternate denser syntax that might make this specific case a bit better.

```yaml
properties:
 projectid: {type: string}
 title: {type: string}
 author: {type: string}
 languages: {type: array, items: {type: string}}
 genre: {type: string}
```

and
```yaml
 credits_so_far: {type: string, readOnly: true}
 modified: {type: string, format: dateTime, readOnly: true}
```

What do you think?

Reply:
Oh, cool! Honestly, I was just building this in the Swagger editor which uses the verbose syntax by default. It looks like it happily supports the denser syntax as well. I'll check out using the denser syntax for some of the schema bits, although I don't want to use it for everything.